traitsui/testing/tester/qt4/implementation/list_editor.py

```diff
@@ -0,0 +1,109 @@
+#  Copyright (c) 2005-2020, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+from traitsui.testing.tester import locator
+from traitsui.testing.tester.registry_helper import find_by_name_in_nested_ui
+
+from traitsui.qt4.list_editor import (
+    CustomEditor,
+)
+
+
+class _IndexedCustomEditor:
+    """ Wrapper for a ListEditor (custom) with an index.
+    """
+
+    def __init__(self, target, index):
+        """
+        Parameters
+        ----------
+        target : CustomEditor
+            The Custom List Editor
+        index : int
+            The index of interest.
+        """
+        self.target = target
+        self.index = index
+
+    @classmethod
+    def from_location(cls, wrapper, location):
+        """ Helper method to create an _IndexedCustomEditor instance.
+
+        Parameters
+        ----------
+        wrapper : UIWrapper
+            UI Wrapper wrapping the Custom List Editor
+        location : instance of locator.Index
+        """
+        return cls(
+            target=wrapper.target,
+            index=location.index,
+        )
```

Comment:
I feel like this method is unnecessary and just makes code harder to read.  I think we might as well just do `solver=lambda wrapper, location: cls(target=wrapper.target, index=location.index)` in line 65